docs/specification.md

```diff
@@ -1,153 +1,207 @@
-    consensus->>+mev_boost: builder_getPayloadHeaderV1
-    mev_boost->>relays: relay_getPayloadHeaderV1
-    Note over mev_boost: select most valuable payload
-    mev_boost-->>-consensus: builder_getPayloadHeaderV1 response
+    consensus->>mev_boost: builder_getHeader
+    mev_boost->>relays: builder_getHeader
+    relays-->>mev_boost: builder_getHeader response
+    Note over mev_boost: verify response matches expected
+    Note over mev_boost: select best payload
+    mev_boost-->>consensus: builder_getHeader response
     Note over consensus: sign the block
-    consensus->>+mev_boost: builder_proposeBlindedBlockV1
+    consensus->>mev_boost: builder_getPayload
     Note over mev_boost: identify payload source
-    mev_boost->>relays: relay_proposeBlindedBlockV1
+    mev_boost->>relays: builder_getPayload
     Note over relays: validate signature
-    relays-->>mev_boost: relay_proposeBlindedBlockV1 response
-    mev_boost-->>-consensus: builder_proposeBlindedBlockV1 response
+    relays-->>mev_boost: builder_getPayload response
+    Note over mev_boost: verify response matches expected
+    mev_boost-->>consensus: builder_getPayload response
 ```
 
-## Required client modifications
+## Structures
 
-- consensus client must implement [blind transaction signing](https://hackmd.io/@paulhauner/H1XifIQ_t#Change-1-Blind-Transaction-Signing)
+### `ExecutionPayloadV1`
 
-## API
+Mirror of [`ExecutionPayloadV1`][execution-payload].
 
-Methods are prefixed using the following convention:
+### `ExecutionPayloadHeaderV1`
 
-- `engine` prefix indicates calls made to the execution client. These methods are specified in the [execution engine APIs](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.8/src/engine/specification.md).
-- `builder` prefix indicates calls made to the mev-boost sidecar.
-- `relay` prefix indicates calls made to a relay.
+Equivalent to `ExecutionPayloadV1`, except `transactions` is replaced with `transactionsRoot`.
+- `parentHash`: `DATA`, 32 Bytes
+- `feeRecipient`:  `DATA`, 20 Bytes
+- `stateRoot`: `DATA`, 32 Bytes
+- `receiptsRoot`: `DATA`, 32 Bytes
+- `logsBloom`: `DATA`, 256 Bytes
+- `prevRandao`: `DATA`, 32 Bytes
+- `blockNumber`: `QUANTITY`, 64 Bits
+- `gasLimit`: `QUANTITY`, 64 Bits
+- `gasUsed`: `QUANTITY`, 64 Bits
+- `timestamp`: `QUANTITY`, 64 Bits
+- `extraData`: `DATA`, 0 to 32 Bytes
+- `baseFeePerGas`: `QUANTITY`, 256 Bits
+- `blockHash`: `DATA`, 32 Bytes
+- `transactionsRoot`: `DATA`, 32 Bytes
 
-### engine_forkchoiceUpdatedV1
+#### SSZ Objects
 
-See [engine_forkchoiceUpdatedV1](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.8/src/engine/specification.md#engine_forkchoiceupdatedv1).
+Consider the following definitions supplementary to the definitions in [`consensus-specs`][consensus-specs].
 
-### builder_proposeBlindedBlockV1
+##### `builder_setFeeRecipientV1` Request
 
-#### Request
+```python
+class Request(Container):
+    feeRecipient: Bytes20
+    timestamp: uint64
+```
```

Comment:
I think we are missing public key and signature here. As seen below:
```
  1. `feeRecipient`: `DATA`, 20 Bytes - Address of account which should receive fees.
  2. `timestamp`: `QUANTITY`, uint64 - Unix timestamp of announcement.
  3. `publicKey`: `DATA`, 48 Bytes - Public key of validator.
  4. `signature`: `DATA`, 96 Bytes - Signature over `feeRecipient`.
  ```
  
Or do we neglect the rest because it's not part of signing root? in that case, `timestamp` should be removed?

Reply:
I'm not sure the best way to go about specify all the different types here. The SSZ types are generally only used to calculate the signing root, so that's why they don't include the public keys or signatures. There is a mistake in the `builder_setFeeRecipientV1` method where it should say the signature is over `feeRecipient` and `timestamp`. I will fix this. I welcome any suggestions on how to make the specification clearer.

> In that case, timestamp should be removed?

Could you clarify why the timestamp should be removed? If `timestamp` is malleable, it could be difficult for a validator to change their fee recipient, because attackers could replay old signatures.

Reply:
> Could you clarify why the timestamp should be removed? If timestamp is malleable, it could be difficult for a validator to change their fee recipient, because attackers could replay old signatures.

Oh no, I don't think it should. I think it's because we missed mentioning `timeStamp` under `signature: DATA, 96 Bytes - Signature over feeRecipient.`, that's why I thought it was not required here. We are good here!